dataprofiler/profilers/profile_builder.py

```diff
@@ -335,11 +339,15 @@ def clean_data_and_get_base_stats(df_series, sample_size,
         na_columns = dict()
         true_sample_list = set()
         total_sample_size = 0
+        sample_ids_used = []
         query = '|'.join(null_values_and_flags.keys())
         regex = f"^(?:{(query)})$"
         
         for chunked_sample_ids in sample_ind_generator:
+            if isinstance(chunked_sample_ids, np.ndarray):
+                chunked_sample_ids = chunked_sample_ids.tolist()
             total_sample_size += len(chunked_sample_ids)
+            sample_ids_used += chunked_sample_ids
```

Comment:
This creates an unnecessary copy increasing memory by columns * rows * int64

Instead you should be able to select the location of the slice and pass that outside, assuming it's not dynamically creating sample_ids. If it is generating sample_ids then this is fine. But I'm not 100% sure when that happens any more anyway